mobilitydb/sql/cbuffer/122_tcbuffer.in.sql

```diff
@@ -0,0 +1,757 @@
+  -- LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE FUNCTION asBinary(tcbuffer, endianenconding text DEFAULT '')
+  RETURNS bytea
+  AS 'MODULE_PATHNAME', 'Temporal_as_wkb'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE FUNCTION asEWKB(tcbuffer, endianenconding text DEFAULT '')
+  RETURNS bytea
+  AS 'MODULE_PATHNAME', 'Tpoint_as_ewkb'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+CREATE FUNCTION asHexEWKB(tcbuffer, endianenconding text DEFAULT '')
+  RETURNS text
+  AS 'MODULE_PATHNAME', 'Temporal_as_hexwkb'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+/******************************************************************************
+ * Constructors
+ ******************************************************************************/
+
+CREATE FUNCTION tcbuffer(cbuffer, timestamptz)
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tinstant_constructor'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE FUNCTION tcbuffer(cbuffer, tstzset)
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tsequence_from_base_tstzset'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE FUNCTION tcbuffer(cbuffer, tstzspan, text DEFAULT 'linear')
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tsequence_from_base_tstzspan'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE FUNCTION tcbuffer(cbuffer, tstzspanset, text DEFAULT 'linear')
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tsequenceset_from_base_tstzspanset'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+/******************************************************************************/
+
+CREATE FUNCTION tcbufferSeq(tcbuffer[], text DEFAULT 'linear',
+    lower_inc boolean DEFAULT true, upper_inc boolean DEFAULT true)
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tsequence_constructor'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE FUNCTION tcbufferSeqSet(tcbuffer[])
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tsequenceset_constructor'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+-- The function is not strict
+CREATE FUNCTION tcbufferSeqSetGaps(tcbuffer[], maxt interval DEFAULT NULL,
+    maxdist float DEFAULT NULL, text DEFAULT 'linear')
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tsequenceset_constructor_gaps'
+  LANGUAGE C IMMUTABLE PARALLEL SAFE;
+
+/******************************************************************************/
+
+CREATE FUNCTION tcbuffer(tgeompoint, tfloat)
+  RETURNS tcbuffer
+  AS 'MODULE_PATHNAME', 'Tcbuffer_constructor'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+
+/******************************************************************************
+ * Cast functions
+ ******************************************************************************/
+
+CREATE FUNCTION tgeompoint(tcbuffer)
+  RETURNS tgeompoint
+  AS 'MODULE_PATHNAME', 'Tcbuffer_to_tgeompoint'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
+CREATE FUNCTION tfloat(tcbuffer)
+  RETURNS tfloat
+  AS 'MODULE_PATHNAME', 'Tcbuffer_to_tfloat'
+  LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
```

Comment:
Are these ideal names? I'd suggest using something like `(t)center` and `(t)radius` instead.

Reply:
In MobilityDB, all casts have their SQL function name as type2(type1) and their C function name as type1_to_type2. If we want to keep this casting, I would recommend keeping the same name if only for uniformity. If we remove the casting indeed, your function names would be great.